src/ripple/shamap/SHAMap.h

```diff
@@ -149,27 +151,31 @@ class SHAMap
             Can throw SHAMapMissingNode
         @note This can cause NodeStore reads
     */
-    // VFALCO NOTE This should return `const&` when SHAMapTreeNode
-    //             stores std::shared_ptr<SHAMapItem const>
-    std::shared_ptr<SHAMapItem const>
-    fetch (uint256 const& key) const;
+    std::shared_ptr<SHAMapItem const> const&
+        fetch (uint256 const& key) const;
 
     // VFALCO NOTE Is "save a copy" the in imperative or indicative mood?
     // save a copy if you only need a temporary
-    std::shared_ptr<SHAMapItem> peekItem (uint256 const& id) const;
-    std::shared_ptr<SHAMapItem> peekItem (uint256 const& id, uint256 & hash) const;
-    std::shared_ptr<SHAMapItem> peekItem (uint256 const& id, SHAMapTreeNode::TNType & type) const;
+    std::shared_ptr<SHAMapItem const> const& peekItem (uint256 const& id) const;
```

Comment:
I mean, address it in the source code by explaining the routine in a comment. And deleting my comment.